caffe2/core/storage.h

```diff
@@ -95,26 +141,43 @@ class StorageImpl {
     } else {
       data_ptr_.reset(src, d);
     }
-    // Sets capacity. If not specified, we will implicitly assume that
-    // the capacity is the current size.
-    if (capacity) {
-      capacity_ = capacity;
-    }
+    capacity_ = capacity;
   }
 
-  // TODO: changed to DataPtr in Aten when shared folder
-  // is ready
-  using DataPtr = std::shared_ptr<void>;
+  void ShareData(const Storage& src) {
+    data_ptr_ = src->data_ptr_;
```

Comment:
This isn't going to work when `DataPtr` becomes a unique pointer like thing. Is this function really the one you wanted?

Reply:
umm.. we can't do `storage_ = src.storage();` since that doesn't keep the old semantics. can DataPtr be shared_ptr?

Reply:
In offline conversation, we concluded that this would be OK if (1) Resize() detaches the storage when it reallocates (which it does) and (2) Storages don't change data_type inplace (which they don't)